### PR TITLE
fix: display programs nav button only if it is configured

### PR DESCRIPTION
**Pull request by dcoa**
**Description**

This PR removes the link of programs from the Header if the service is not configured.

It aims to solve the following issue https://github.com/openedx/frontend-app-learner-dashboard/issues/372

**How to test**

   1. Create a instance with frontend-app-learner-dashobord mounted in the current branch.
   1. Launch the instance and open it in the browser.
   1. Access the learner dashboard.
   1. By default you should not see nav button.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dcoa!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@2U-aperture`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Review by deborahgu — APPROVED** [review]
this looks good to me. If we weren't time constrained I would want to add the test for a unpermissioned user, just because that's what broke last time, but I doubt we have a great way to do that quickly, and this shouldn't remotely have a chance of running into that problem. Thanks for the quick turnaround on the fix!

File: src/config/index.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const configuration = {
   LOGO_URL: process.env.LOGO_URL,
   ENABLE_EDX_PERSONAL_DASHBOARD: process.env.ENABLE_EDX_PERSONAL_DASHBOARD === 'true',
   SEARCH_CATALOG_URL: process.env.SEARCH_CATALOG_URL || null,
+  ENABLE_PROGRAMS: !!process.env.ENABLE_PROGRAMS,
```
Comment (arbrandes) [inline src/config/index.js:22]: Can you please check for the `'true'` string explicitly, like in line 20 above?  There's a reason for it: `.env` files only support strings, not booleans.  (Yes, I know, there are some boolean checks that don't respect this - they're basically wrong.)

That leads me to a second request: can you please add `ENABLE_PROPRAMS` to the existing `.env` files?

File: src/containers/LearnerDashboardHeader/index.test.jsx [hunk 2]
```diff
@@ -39,4 +39,9 @@ describe('LearnerDashboardHeader', () => {
     const wrapper = shallow(<LearnerDashboardHeader />);
     expect(wrapper.instance.findByType(Header)[0].props.secondaryMenuItems.length).toBe(1);
   });
+  test('should display Programs link if the service is configured in the backend', () => {
```
Comment (arbrandes) [inline src/containers/LearnerDashboardHeader/index.test.jsx:42]: Looks like we need to reword the test title, here.

```suggestion
  test('should display Programs link if it's enabled by configuration', () => {
```

**Review by arbrandes — CHANGES_REQUESTED** [review]
Thanks for doing this!  Just a couple of changes, if you don't mind.

File: src/config/index.js (hunk 1 above)
Comment (dcoa) [reply src/config/index.js]: I addressed both comments :) 

@arbrandes

**Review by arbrandes — APPROVED** [review]
Thank you @dcoa, looks great!

**Comment by deborahgu** [comment]
for what it's worth I verified this again with your changes, and it still works great, so merge whenever you are ready.

**Comment by dcoa** [comment]
Thank you!!
I don't have write access inn this repo, could any of you help me with the merge @deborahgu , @arbrandes ?

**Comment by deborahgu** [comment]
I've got you, @dcoa! 😃

**Pull request merged**